src/serial.rs

```diff
@@ -377,11 +380,20 @@ impl<T: Trigger, EV: SerialEvents, W: Write> Serial<T, EV, W> {
                         self.received_data_interrupt().map_err(Error::Trigger)?;
                     }
                 } else {
-                    self.out.write_all(&[value]).map_err(Error::IOError)?;
-                    self.out.flush().map_err(Error::IOError)?;
-                    self.serial_evts.out_byte();
-
+                    let res = self
+                        .out
+                        .write_all(&[value])
+                        .map_err(Error::IOError)
+                        .and_then(|_| self.out.flush().map_err(Error::IOError))
+                        .map(|_| self.serial_evts.out_byte())
+                        .map_err(|err| {
+                            self.serial_evts.tx_lost_byte();
+                            err
+                        });
+                    /// Because we cannot block the driver, the THRE interrupt is sent
+                    /// irrespective of weather we are able to write the byte or not
```

Comment:
```suggestion
                    // Because we cannot block the driver, the THRE interrupt is sent
                    // irrespective of whether we are able to write the byte or not.
```